department-of-veterans-affairs/vets-website
# MHV-62497 Failure-Of-PHR-Refresh-Status-Check-Is-Not-Handled

## Summary
Update `NewRecordsIndicator` to include `CALL_FAILED` to handle the failure status of API calls.

## Related issue(s)

[MHV-62497](https://jira.devops.va.gov/browse/MHV-62497) - Failure of PHR refresh status check is not handled 

Failure of PHR refresh status check is not handled

Currently we are not handling the failure of the /status API call. Identify what happens if this status check fails, then come up with a plan to remediate this.

I started a discussion about this here: https://dsva.slack.com/archives/C04ER7MHX5E/p1729180324631989



User Story: As a Medical Records team, we want to be certain we are handling the failure of the /status API calls so that we can plan a remediation for this issue.

GIVEN

WHEN

THEN

Feature Flag N

DataDog/Analytics N

Manual Testing - Y  

Automated Testing Y/N ?  

Accessibility Testing N 

UCD Validation Y/N

 

Acceptance Criteria:

AC1  Log this issue in the playbook

AC2 PHR refresh failure status is handled correctly

AC3
AC4
AC5

Links to UCD:

Figma Link:

User Flow:

Mobile:

Desktop:

Other Design Notes:

Architectural Notes: Always check for the latest designs in Figma

Other Design Notes:
## Testing done
Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

## Screenshots
![Screenshot 2024-11-20 at 9 09 24 AM (3)](https://github.com/user-attachments/assets/3b35aa2e-d3b7-44f3-87be-60fc2aae7836)


## What areas of the site does it impact?
The modification affects all API calls  in Medical Records pages.

## Acceptance criteria
Inclusion of CALL_FAILED Status:

1. The NewRecordsIndicator must be updated to include a new status called CALL_FAILED.
This status will be used to represent API calls that have failed.

Correct Handling of API Failures:

2. When an API call fails, the CALL_FAILED status must be triggered in the NewRecordsIndicator.
The CALL_FAILED status should be applied only in cases where the API call returns an error or fails to complete successfully (e.g., network errors, server errors).

UI Updates:

3. The UI should display the CALL_FAILED status appropriately, either by showing an error message or updating the indicator to reflect the failure state.
The CALL_FAILED status should be distinguishable from other statuses (e.g., by using an error icon or a red color to indicate failure).

### Quality Assurance & Testing


### Authentication


## Summary
Update lkfja)

[MHV-61246](https://jira.devops.va.gov/browse/MHV-61246) - Add the "Last updated" text to the print view for List pages

Add the "Last updated" text to the print view for List pages

The NewRecordsIndicator has been removed from the print view, because it was printing the spinner, but we probably want to print something there.

Work with UCD to decide how it should look when printed in various scenarios, then implement inside NewRecordsIndicator.

 

User Story:  As a MR user, I want to    so that I can

 

GIVEN:

WHEN:

THEN:

 

Feature Flag Y/N ?

DataDog Analytics  Y/N ?

Manual Testing Y/N ? 

Automated Testing Y/N ?

Accessibility Testing Y/N ?

UCD Validation Y/N

 

Acceptance Criteria:

AC1

AC2

AC3

AC4

Links to UCD:

Figma Link:

Other Design Notes:
## Testing done
Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

## Screenshots




## What areas of the site does it impact?
The modification affects the Vitals Detail page in Medical Records pages.

## Acceptance criteria
The focus shifts back to the title when pagination navigates to the first page. 

### Quality Assurance & Testing


### Authentication
